### PR TITLE
PLAT-870 Some Cleanup of TypeScript Code

### PR DESCRIPTION
**Pull request by oliver-richers**
- renamed `_DOM_CONTEXT_` to `AJAX_ENGINE`
- renamed constant `DOM_POPUP_ENGINE` to `POPUP_ENGINE`
- renamed method `setMaximumZIndex` to `raise`
- renamed some JavaScript and TypeScript files
- merged `Utils.ts` into `AjaxEngine.ts`
- extracted `Facade.ts` from `AjaxEngine.js`
- removed constant `c`

This is not the end of the story, but a massive improvement, IMO:

image
image

### CODE REVIEW

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/event/DomEvent.ts [hunk 1]
```diff
@@ -1,8 +1,31 @@
+	
+function listenToDomEvent(nodeID: number, event: string, doListen: boolean) {
```
Comment (alexander-schmidt) [inline platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/event/DomEvent.ts:2]: `nodeID` -> `nodeId`

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/AjaxEngine.ts [hunk 2]
```diff
@@ -97,87 +96,19 @@ class AjaxEngine {
...
-		}
+
+	/**
+	 * Pushes the given URL into the history of the browser.
+	 */
+	public pushBrowserHistoryState(page: string, url: string) {
+	    history.pushState({page: page}, "", url);
 	}
 
+
```
Comment (alexander-schmidt) [inline platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/AjaxEngine.ts:107]: surplus newline

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/event/DomEvent.ts (hunk 1 above)
Comment (oliver-richers) [reply platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/event/DomEvent.ts]: fixed

File: platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/AjaxEngine.ts (hunk 2 above)
Comment (oliver-richers) [reply platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/AjaxEngine.ts]: fixed

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**